cmd/kueuectl/app/stop/stop_clusterqueue.go

```diff
@@ -0,0 +1,159 @@
+
+	genericiooptions.IOStreams
+}
+
+func NewClusterQueueOptions(streams genericiooptions.IOStreams) *ClusterQueueOptions {
+	return &ClusterQueueOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("stop").WithTypeSetter(scheme.Scheme), // TODO check what print flags are needed
+		IOStreams:  streams,
+	}
+}
+
+func NewClusterQueueCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewClusterQueueOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:                   "clusterqueue NAME [--dry-run STRATEGY] [--keep-already-running] [--resource-flavor=rfnam]",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"cq"},
+		Short:                 "Stop the ClusterQueue",
+		Long:                  cqLong,
+		Example:               cqExample,
+		Args:                  cobra.MatchAll(cobra.ExactArgs(1), cobra.OnlyValidArgs),
+		Run: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(o.Complete(clientGetter, cmd, args))
+			cobra.CheckErr(o.Run(cmd.Context()))
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	addKeepAlreadyRunningFlagVar(cmd, &o.KeepAlreadyRunning)
+
+	return cmd
+}
+
+// Complete completes all the required options
+func (o *ClusterQueueOptions) Complete(clientGetter util.ClientGetter, cmd *cobra.Command, args []string) error {
+	o.ClusterQueueName = args[0]
+
+	clientset, err := clientGetter.KueueClientSet()
+	if err != nil {
+		return err
+	}
+
+	o.Client = clientset.KueueV1beta1()
+
+	o.DryRunStrategy, err = util.GetDryRunStrategy(cmd)
+	if err != nil {
+		return err
+	}
+
+	// TODO: ask how it should work with dry-run (check if it should check if clusterqueue can be stopped)
+	err = util.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
+	if err != nil {
+		return err
+	}
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	// TODO: should it at least print that clusterqueue was scheduled to be stopped?
+	o.PrintObj = printer.PrintObj
+
+	return nil
+}
+
+// Run executes the command
+func (o *ClusterQueueOptions) Run(ctx context.Context) error {
+	cq, err := o.Client.ClusterQueues().Get(ctx, o.ClusterQueueName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if cq == nil {
+		return nil
+	}
+
+	cqOriginal := cq.DeepCopy()
+	// TODO: ask what should happen if Hold/HoldAndDrain is in progress, and someone tries to start it again
+	// TODO: with inactive(/any other then existing) stop policy - it can be the only status that can be started
+	o.stopClusterQueue(cq)
+
+	opts := metav1.PatchOptions{}
+	patch := client.MergeFrom(cqOriginal)
+	data, err := patch.Data(cq)
+	if err != nil {
+		return err
+	}
+	cq, err = o.Client.ClusterQueues().
+		Patch(ctx, o.ClusterQueueName, types.MergePatchType, data, opts)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *ClusterQueueOptions) stopClusterQueue(cq *v1beta1.ClusterQueue) {
+	if o.KeepAlreadyRunning {
+		cq.Spec.StopPolicy = ptr.To(v1beta1.Hold)
```

Comment:
What happens to workloads that are waiting for admission check but are not yet fully started? Are they considered running or not running?

cc: @trasc

Reply:
Yes, the workloads that are only reserving louse the reservation immediately.